### DISCUSSION TITLE
How to get form data in beforeCreate hook

### DISCUSSION THREAD

**Discussion by mohandtharsha** [filamentphp/filament#9323] [Help]
I want to customize the creating functionality if is_featured is true then i have to show popup modal with subscription plans but I am finding difficulty to achieve that can someone help me on this?

Below is my code,

[<?php

namespace App\Filament\Employer\Resources\JobsResource\Pages;

use App\Filament\Employer\Resources\JobsResource;
use Filament\Actions;
use Filament\Resources\Pages\CreateRecord;
use Filament\Notifications\Notification;
use Filament\Notifications\Actions\Action;
use Filament\Forms\Form;

class CreateJobs extends CreateRecord
{
    protected static string $resource = JobsResource::class;

    protected function getRedirectUrl(): string
    {
        return $this->getResource()::getUrl('index');
    }

    protected function getCreatedNotification(): ?Notification
    {
        return Notification::make()
            ->success()
            ->title('Saved')
            ->body('The Job has been created successfully.');
    }

    protected function beforeCreate(Form $form): void
    {
        $formData = $form->data;
        dd($formData);
        if (!$this->getRecord()) {
            Notification::make()
                ->warning()
                ->title('You don\'t have an active subscription!')
                ->body('Choose a plan to continue.')
                ->persistent()
                ->actions([
                    Action::make('subscribe')
                        ->button()
                        // ->url(route('subscribe'), shouldOpenInNewTab: true),
                ])
                ->send();
        
            $this->halt();
        }
    }
}

**Comment by VincvanAs** [comment]
You want to show the modal on page load or when a user save a form?

**Comment by mohandtharsha** [comment]
Yeah i Want whenever user submits the form need a popup with packages information like package1,package2 where they can select

**Reply by leandrocfe** [reply]
You are creating a record, right? Why not a select in your form?

**Comment by yogesh1663** [comment]
you can use $data = $this->data;
$name = $data['name];
to get form data

protected function beforeCreate(): void
{
    $formData = $this->data;
    dd($formData);
    now apply your logic to halt the process
}
